File: infullmvp/src/main/java/com/infullmobile/android/infullmvp/InFullMvpBottomSheetDialogFragment.java
package com.infullmobile.android.infullmvp;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.CallSuper;
import android.support.design.widget.BottomSheetDialogFragment;
import android.view.MenuItem;
import android.view.View;

public abstract class InFullMvpBottomSheetDialogFragment<
        PresenterType extends Presenter<PresentedViewType>,
        PresentedViewType extends PresentedDialogView<PresenterType>
        > extends BottomSheetDialogFragment {

    protected abstract PresenterType getPresenter();

    protected abstract PresentedViewType getPresentedView();

    protected abstract void injectIntoGraph();

    @CallSuper
    @Override
    public void setupDialog(Dialog dialog, int style) {
        super.setupDialog(dialog, style);
        injectIntoGraph();
        dialog.setContentView(
                View.inflate(getContext(), getPresentedView().getLayoutResId(), null)
        );
        getPresentedView().bindUiElements(dialog, getPresenter());
        getPresenter().bind(
                getArguments() != null ? getArguments() : new Bundle(),
                new Bundle(),
                getActivity().getIntent().getData()
        );
    }

    @Override
    public boolean onContextItemSelected(final MenuItem item) {
        return getPresenter().onContextItemSelected(item) || super.onContextItemSelected(item);
    }
}

File: infullmvp-kotlin/src/main/java/com/infullmobile/android/infullmvp/InFullMvpBottomSheetDialogFragment.kt
package com.infullmobile.android.infullmvp

import android.app.Dialog
import android.os.Bundle
import android.support.design.widget.BottomSheetDialogFragment
import android.view.MenuItem
import android.view.View

abstract class InFullMvpBottomSheetDialogFragment<
        PresenterType : Presenter<PresentedViewType>,
        out PresentedViewType : PresentedDialogView<PresenterType>
        > : BottomSheetDialogFragment() {

    abstract val presenter: PresenterType
    abstract val presentedView: PresentedViewType

    abstract fun injectIntoGraph()

    override fun setupDialog(dialog: Dialog, style: Int) {
        super.setupDialog(dialog, style)
        injectIntoGraph()
        val rootView = View.inflate(context, presentedView.layoutResId, null)
        dialog.setContentView(rootView)
        presentedView.bindUiElements(dialog, presenter)
        presenter.bind(arguments?: Bundle(), Bundle(), activity!!.intent.data)
    }

    override fun onContextItemSelected(item: MenuItem?): Boolean {
        return presenter.onContextItemSelected(item) ||  super.onContextItemSelected(item)
    }
}